packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -1,72 +1,100 @@
-import {
-  GraphQLClient,
-  GraphQLRequestClient,
-  getAppRootId,
-  SearchServiceConfig,
-  SearchQueryService,
-} from '@sitecore-jss/sitecore-jss/graphql';
+import { GraphQLClient, GraphQLRequestClient, PageInfo } from '@sitecore-jss/sitecore-jss/graphql';
 import { debug } from '@sitecore-jss/sitecore-jss';
 
-/** @private */
-export const queryError =
-  'Valid value for rootItemId not provided and failed to auto-resolve app root item.';
-
 /** @private */
 export const languageError = 'The list of languages cannot be empty';
 
-// Even though _hasLayout should always be "true" in this query, using a variable is necessary for compatibility with Edge
+const languageEmptyError = 'The language must be a non-empty string';
+
 const defaultQuery = /* GraphQL */ `
-  query SitemapQuery(
-    $rootItemId: String!
-    $language: String!
-    $pageSize: Int = 10
-    $hasLayout: String = "true"
-    $after: String
-  ) {
-    search(
-      where: {
-        AND: [
-          { name: "_path", value: $rootItemId, operator: CONTAINS }
-          { name: "_language", value: $language }
-          { name: "_hasLayout", value: $hasLayout }
-        ]
-      }
-      first: $pageSize
-      after: $after
-    ) {
-      total
-      pageInfo {
-        endCursor
-        hasNext
-      }
-      results {
-        url {
-          path
+query SitemapQuery(
+  $siteName: String!,
+  $language: String!,
+  $includedPaths: String[],
+  $excludedPaths: String[],
+  $pageSize: Int = 10,
+  $after: String
+) {
+  site {
+    siteInfo(site: $siteName) {
+      routes(
+        language: $language
+        includedPaths: $includedPaths
+        excludedPaths: $excludedPaths
+        first: $pageSize
+        after: $after
+      ){
+        total
+        pageInfo {
+          endCursor
+          hasNext
+        }
+        results: routesResult{
+          path: routePath 
         }
       }
     }
   }
+} 
 `;
-
 /**
- * The schema of data returned in response to a page list query request
+ * type for input variables for the site routes query
  */
-export type PageListQueryResult = { url: { path: string } };
+interface SiteQueryVariables {
```

Comment:
Include "Route" in the name?
```suggestion
interface SiteRouteQueryVariables {
```